File: base_core/src/main/java/com/alvin/base_core/base/livedata/EventLiveData.java
package com.alvin.base_core.base.livedata;

/**
 * 解决数据倒灌
 * https://github.com/KunMinX/UnPeek-LiveData
 */
public class EventLiveData<T> extends ProtectedUnPeekLiveData<T> {

    public EventLiveData() {
        super();
    }

    @Override
    public void setValue(T value) {
        super.setValue(value);
    }

    @Override
    public void postValue(T value) {
        super.postValue(value);
    }

    public static class Builder<T> {

        /**
         * 是否允许传入 null value
         */
        private boolean isAllowNullValue;

        public Builder<T> setAllowNullValue(boolean allowNullValue) {
            this.isAllowNullValue = allowNullValue;
            return this;
        }

        public EventLiveData<T> create() {
            EventLiveData<T> liveData = new EventLiveData<>();
            liveData.isAllowNullValue = this.isAllowNullValue;
            return liveData;
        }
    }
}

File: base_core/src/main/java/com/alvin/base_core/base/livedata/EventLiveData.kt
package com.alvin.base_core.base.livedata

/**
 * <h3> 作用类描述：包装类</h3>
 *
 * @Package :        com.yleanlink.mvvm.base.livedata
 * @Date :           2021/12/20-11:45
 * @author 高国峰
 */
open class EventLiveData<T>(isAllowNullValue: Boolean = true) :
    ProtectedUnPeekLiveData<T>(isAllowNullValue)